### PR TITLE
Properly implement D3DDevice_GetDisplayFieldStatus patch

### PR DESCRIPTION
**Pull request by LukeUsher**
This also fixes an issue where the default eeprom was not created correctly

### CODE REVIEW

**Comment by LukeUsher** [comment]
I'd be interested to see if this properly solves the Azurik regression reported here: https://github.com/Cxbx-Reloaded/game-compatibility/issues/444

**Comment by blueshogun96** [comment]
I'm sorry, but for me, the issue persists, even with this new commit.

Shogun

File: src/Common/EmuEEPROM.cpp [hunk 1]
```diff
@@ -121,18 +121,28 @@ xboxkrnl::XBOX_EEPROM *CxbxRestoreEEPROM(char *szFilePath_EEPROM_bin)
...
 	// TODO : Verify checksums
 
+	// Check for (and fix) invalid fields that were set by previous versions of Cxbx-Reloaded
+	// Without this, all users would have to delete their EEPROM.bin
+	// The issue was that the AV_FLAG_XXhz was set in the wrong field, we fix it by
+	// resetting FactorySettings.AVRegion and setting user video flags to the default
+	// The user can then set their desired settings using the Xbox Dashboard
+	if (pEEPROM->FactorySettings.AVRegion == AV_STANDARD_NTSC_M) {
+		pEEPROM->UserSettings.VideoFlags = 0;
+		pEEPROM->FactorySettings.AVRegion = AV_STANDARD_NTSC_M | AV_FLAGS_60Hz;
```
Comment (PatrickvL) [inline src/Common/EmuEEPROM.cpp]: Maybe add a log message this had happened

File: src/CxbxKrnl/EmuKrnlAv.cpp [hunk 2]
```diff
@@ -63,16 +63,40 @@ PVOID g_pPersistedData = NULL;
...
+	result = xboxkrnl::ExQueryNonVolatileSetting(
+		xboxkrnl::XC_VIDEO,
+		&type,
+		&userSettings,
+		sizeof(ULONG),
+		&resultSize);
+
+	// If this failed, default to no user-options set
+	if (result != STATUS_SUCCESS || resultSize != sizeof(ULONG)) {
+		userSettings = 0;
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuKrnlAv.cpp]: Maybe add a message here too

File: src/CxbxKrnl/EmuKrnlAv.cpp [hunk 3]
```diff
@@ -63,16 +63,40 @@ PVOID g_pPersistedData = NULL;
...
+	NTSTATUS result = xboxkrnl::ExQueryNonVolatileSetting(
+		xboxkrnl::XC_FACTORY_AV_REGION,
+		&type,
+		&avRegion,
+		sizeof(ULONG),
+		&resultSize);
+
+	// If this failed, default to AV_STANDARD_NTSC_M | AV_FLAGS_60Hz
+	if (result != STATUS_SUCCESS || resultSize != sizeof(ULONG)) {
+		avRegion = AV_STANDARD_NTSC_M | AV_FLAGS_60Hz;
```
Comment (PatrickvL) [inline src/CxbxKrnl/EmuKrnlAv.cpp]: And here

**Comment by jarupxx** [comment]
Good for me, solves the Azurik regression.

File: src/CxbxKrnl/EmuKrnlAv.cpp (hunk 2 above)
Comment (LukeUsher) [reply src/CxbxKrnl/EmuKrnlAv.cpp]: I don't think this case (and the other avRegion case) need any logging as this is the same behavior as the real xbox, so it's a situation that can happen, even on real hardware.

**Comment by PatrickvL** [comment]
I've merged this, based on jarpuxx test results

**Comment by Sarkie** [comment]
Could we have a repo where we can dump our system info?

Like the fact it works for one person and not the other.
Could be easier to spot why.

**Comment by LukeUsher** [comment]
I managed to get test results from Azurik NTSC: It seems to behave differently in both release and debug builds. In release builds, it seems to crash, while in Debug builds, it boots and is able to play audio.

This leads me to the belief that it there must be some sort of race-condition that isn't being hit on slower machines, and the debug build works for some users due to the lack of compiler optimization.

**Comment by PatrickvL** [comment]
Another explanation could be different allocation patterns due to logging; Because currently Xbox and host allocations might still go the same heap, code that access buffer bounds could damage functionality

**Pull request merged**